lib/github-pages-health-check.rb

```diff
@@ -65,11 +66,15 @@ def valid_domain?
       PublicSuffix.valid? domain
     end
 
-    # Is this domain an SLD, meaning a CNAME would be innapropriate
+    # Is this domain an apex domain, meaning a CNAME would be innapropriate
     def apex_domain?
-      PublicSuffix.parse(domain).trd == nil
-    rescue
-      false
+      return @apex_domain if defined?(@apex_domain)
+
+      answers = Resolv::DNS.open { |dns|
+        dns.getresources(absolute_domain, Resolv::DNS::Resource::IN::NS)
```

Comment:
Ahh. Makes sense. Last thought would be that we might be able to continue to use Net:Dns (if there's any advantage over Resolve (?) with something like:

```ruby
Net::DNS::Resolver.start("google.com", Net::DNS::NS)
```

You're the true Rubyist here, so trust your judgement on the dependency. Feel free to merge when you're ready. Thanks for tackling this. Gonna make many users :smile_cat: .